Shiny App Indicator

steffilazerte opened the pull request:
This is the shiny app indicator for protected areas

Issues to fix in Shiny App:
- [ ] In line graph, NAs should be 0s
- [ ] Interactivity of bottom panel acceptable?
- [ ] Is extra space in app acceptable (currently there to allow a slightly taller barchart)?
    - Should barchart be moved elsewhere? 
    - Should barchart be combined with BEC Zones barchart, below?
- [ ] Prevent transparency when loading (cf. MSW app)
- [ ] Confirm that colours are good
- [ ] Any other formatting?

Issues to fix in Data:
- [ ] Merge spatial layers to find more data on date of protection
- [ ] Need to further simplify certain ecoregions (slow to load)

Outstanding issues:
- [ ] BEC Zones figure

steffilazerte commented:
Here are some screenshots of how it looks.
The one problem I see is the amount of white space, but perhaps you'll need to test it locally to see how much of a problem that is.

![screenshot1_2021-03-03](https://user-images.githubusercontent.com/14676081/109843828-1e535c80-7c11-11eb-8529-f30c927f4fbe.png)
![screenshot2_2021-03-03](https://user-images.githubusercontent.com/14676081/109843830-201d2000-7c11-11eb-9d47-8667ec627c59.png)

ateucher commented:
Looking good so far @steffilazerte! I haven't had a chance to review fully, but a couple of notes:

- The code ran for me no problem first time!
- i think the line chart should be cumulative
- In the app, clicking on the two outermost marine ecoregions, the polygons look very simple and blocky. Do you know if they are actually that shape? It looks a bit like they have been oversimplified by `rmapshaper::ms_simplify()`
- Can the ecoregion bar chart be sorted by amount protected?
- For the marine colours, I wonder if we should flip them so dark blue is PPA so they mirror the terrestrial palette?
- I wonder if there's a way to link the map and the bar chart on hover? It's hard to know which ecoregion is which otherwise.  Acknowledging that it's tricky when the click event shows the inset map in place of the chart, I wonder if we can be creative about another panel or something?

KarHarker commented:
Hi Steffi, this is looking great! Thanks for all your hard work. 
I just had a few comments to add onto Andy's:
- For the barchart with all ecoregions: could you change the x-axis to percent of ecoregion protected? That way we can control for the size of ecoregion
- Group ecoregion barchart by marine and land and then order by percent protected (within marine and land categories)

Andy and I revisited the issue you brought up earlier with linking the ecoregion on the map with the barchart because the barchart was being replaced with a inset map. We think that the hover link would be pretty useful to have so we were wondering if it would work to have the inset maps be a pop out feature from the main map instead of replacing the barchart in the right panel. We aren't sure if this can be done in ggiraph though - do you have any thoughts on that?

steffilazerte commented:
I'll address the other comments soon, but first wanted to discuss the layout issue and get that figured out. 

I don't think we can do a popout (I'm assuming you mean something like a large tooltip with a figure in it?). It's not to say we could do it at all, but I think that even if we could, it would be fragile and I'm not sure it's worth it. 

Here are some alternative ideas:

1. One option is to move the map next to the line/barcharts. Here I've zoomed out to include everything, but in a regular browser you'd scroll up and down. I would be able to link the hovered regions to the bars in this case and users could click on a bar OR on a region to change the bottom half. 

![layout_idea1](https://user-images.githubusercontent.com/14676081/110546402-076ea780-80f4-11eb-87f6-393839947f1b.png)

2. Also, if I do go to a cumulative graph, we *might* be able to get away with removing the overall barplot (we're still missing some dates, although not nearly as many). 
![layout_idea2](https://user-images.githubusercontent.com/14676081/110548043-e5762480-80f5-11eb-8859-6d7b5152c687.png)

3. Finally we could make two small shiny apps? With text between them? One bc map / ecoregions barchart and one bc map / subregion map / subregion line graph and bar chart. The only thing would be trying to make the bc map be informative in a different manner and not just duplicate the data. Possibly by having the first one show shading according to the amount protected? We could also add in the bec zones maps/barchart here if that would fit.

My preference would be for 1 or 2. I think they're a bit more straightforward, but I'm happy to give 3 a shot if you think that'd be best.

steffilazerte commented:
I've been playing with having the secondary plots below, but it does mean scrolling to look at them. I'm not sure what's worse.Ah! Came up with another idea. What about something like option 1 or 2, but where the bottom panels *replace* the top panels, and we could have a little button or something to return to the big map?

steffilazerte commented:
Okay, at the risk of overwhelming you guys with a flurry of different layout ideas, I went ahead with the original layout and managed to get the selection issue figured out, and fixed the flickering. The only problem is that because the map of BC aligns with axes of the second figure, it changes size a bit when going from the bar plot to the subregion plots (but not when switching between subregions). 

This app is the current version in the pull request. Screen shots below:

![layout_idea3](https://user-images.githubusercontent.com/14676081/110665582-21f55f00-818e-11eb-9365-a908d04dbfec.png)

**Note** 
- the bar chart still needs work, the proportions aren't correct
- figure below is cumulative, and I added the totals (including regions without dates) to the end of the line with a dotted line to indicate that it includes missing dates.

steffilazerte commented:
Hmm, you'll definitely have to re-run the 04_analysis.R, and, unfortunately, you might have to rerun 03_summarize.R (which takes a while). However, I *think* the changes for 03_summarize mostly deal with where data got saved? So you might be able to get away with just moving some of the final data files to the "out" folder.

Sorry, I've been a bit haphazard in what I'm working on, as I find things to do while bec zones are running (grrrr! but finally done)

ateucher commented:
Sorry it took so long to get back to you @steffilazerte  - it looks great! I ticked off a few boxes in the original comment above.

A final few suggestsions/question:

Bottom line graph:
1. Right now the marine areas have bar charts - they should be the same as the terrestrial areas.
2. Can we do percentage instead of total area?
3. Can we make it a stacked area chart so the PPA area is on the bottom, and the OECMs are stacked on top, and the total height of the area is the total amount protected? 
4. Before any ecoregion is clicked, can we have provincial total area chart on the bottom, to hold the space?
5. Can the dark green/blue be the PPA, and the light green/blue be the OECMs?

Map:
1. Do you think we can make the size constant so when the bar chart is replaced with the ecoregion map, it doesn't shrink? It's ok if the inset ecoregion map ends up being a bit smaller because of this. I also think if there is a provincial line graph at the bottom (before it's clicked as in #3 above), it will help hold the space constant...
2. Can the polygons be shaded by the percent protected (more = darker)?

Bar char
1. As in the line graph, can the dark green/blue be the PPA, and the light green/blue be the OECMs?
2. Grouping by marine/terrestrial (already in your checklist)
3. The axis label should be `%`, not `ha`

Finally, do you think it's possible to add a button/link to "reset" so that it can easily be set back to the BC-wide view? I figured out that if you click the already-selected ecoregion it does this, but it's not 100% intuitive that it will do that.

ateucher commented:
Looking so good @steffilazerte. I think once you've done the final few checklist items above, we're pretty much done!

A couple of other small formatting requests:
1. Can you put axis lines on the bottom area plots? On the facetted provincial plots, it's not immediately obvious they are two different plots. I think a vertical axis line (plus maybe just a touch more space between them?) would separate them a bit more...
2. For the hover tips on the bottom plots: Can the text separate out the two types of protection? Eg.,

**Year**: 1987
**Parks and Protected Areas**: 18%
**OECM**: 12%

3. Similar to #2, can you also break out the PPA and the OECM percentages in the bar chart tooltips?

**Ecoregion**: Ecoregion Name
**Parks and Protected Areas**: 18%
**OECM**: 12%
